sklearn/gaussian_process/gpc.py

```diff
@@ -0,0 +1,707 @@
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        if self.kernel is None:  # Use an RBF kernel as default
+            self.kernel_ = \
+                C(1.0, c_bounds="fixed") * RBF(1.0, l_bounds="fixed")
+        else:
+            self.kernel_ = clone(self.kernel)
+
+        self.X_train_ = np.copy(X) if self.copy_X_train else X
+
+        # Encode class labels and check that it is a binary classification
+        # problem
+        label_encoder = LabelEncoder()
+        self.y_train_ = label_encoder.fit_transform(y)
+        self.classes_ = label_encoder.classes_
+        if self.classes_.size > 2:
+            raise ValueError("%s supports only binary classification. "
+                             "y contains classes %s"
+                             % (self.__class__.__name__, self.classes_))
+        elif self.classes_.size == 1:
+            raise ValueError("{0:s} requires 2 classes.".format(
+                self.__class__.__name__))
+
+        if self.kernel_.n_dims == 0:  # no tunable hyperparameters
+            pass
+        elif self.optimizer is not None:
+            # Choose hyperparameters based on maximizing the log-marginal
+            # likelihood (potentially starting from several initial values)
+            def obj_func(theta, eval_gradient=True):
+                if eval_gradient:
+                    lml, grad = self.log_marginal_likelihood(
+                        theta, eval_gradient=True)
+                    return -lml, -grad
+                else:
+                    return -self.log_marginal_likelihood(theta)
+
+            # First optimize starting from theta specified in kernel
+            optima = [(self._constrained_optimization(obj_func,
+                                                      self.kernel_.theta,
+                                                      self.kernel_.bounds))]
+
+            # Additional runs are performed from log-uniform chosen initial
+            # theta
+            if self.n_restarts_optimizer > 1:
+                if not np.isfinite(self.kernel_.bounds).all():
+                    raise ValueError(
+                        "Multiple optimizer restarts (n_restarts_optimizer>1) "
+                        "requires that all bounds are finite.")
+                bounds = self.kernel_.bounds
+                for iteration in range(1, self.n_restarts_optimizer):
+                    theta_initial = np.exp(self.rng.uniform(bounds[:, 0],
+                                                            bounds[:, 1]))
+                    optima.append(
+                        self._constrained_optimization(obj_func, theta_initial,
+                                                       bounds))
+            # Select result from run with minimal (negative) log-marginal
+            # likelihood
+            self.kernel_.theta = \
+                optima[np.argmin(map(itemgetter(1), optima))][0]
```

Comment:
Could you do this in two steps, e.g. `function_values = map(itemgetter(1), optima)` or `function_values = [opt[1] for opt in optima]` followed by `self.kernel_.theta = optima[np.argmin(function_values)][0]`?